### ISSUE TITLE
Circular require?

### ISSUE THREAD

**Issue by lilith** [RubyMoney/monetize#105]
warning: loading in progress, circular require considered harmful - /home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/monetize.rb

```
/home/n/.rvm/rubies/ruby-2.4.1/bin/ruby -w -I"lib:test" -I"/home/n/.rvm/gems/ruby-2.4.1/gems/rake-12.1.0/lib" "/home/n/.rvm/gems/ruby-2.4.1/gems/rake-12.1.0/lib/rake/rake_test_loader.rb" "test/test_account.rb" 
/home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/collection.rb:3: warning: loading in progress, circular require considered harmful - /home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/monetize.rb
	from /home/n/.rvm/gems/ruby-2.4.1/gems/rake-12.1.0/lib/rake/rake_test_loader.rb:5:in  `<main>'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/rake-12.1.0/lib/rake/rake_test_loader.rb:5:in  `select'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/rake-12.1.0/lib/rake/rake_test_loader.rb:17:in  `block in <main>'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/rake-12.1.0/lib/rake/rake_test_loader.rb:17:in  `require'
	from /home/n/Documents/imazen/ks-deliver/test/test_account.rb:3:in  `<top (required)>'
	from /home/n/Documents/imazen/ks-deliver/test/test_account.rb:3:in  `require_relative'
	from /home/n/Documents/imazen/ks-deliver/ks.rb:6:in  `<top (required)>'
	from /home/n/Documents/imazen/ks-deliver/ks.rb:6:in  `require'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/monetize.rb:7:in  `<top (required)>'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/monetize.rb:7:in  `require'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/collection.rb:3:in  `<top (required)>'
	from /home/n/.rvm/gems/ruby-2.4.1/gems/monetize-1.7.0/lib/collection.rb:3:in  `require'
```

**Comment by antstorm** [comment]
@nathanaeljones this is quite odd since `lib/collection.rb` only requires `forwardable`. I don't see any circular references here. Can you please post a sample of the code you're calling this from?

**Comment by snowblink** [comment]
@nathanaeljones does 1.8.0 fix this for you? #100 may have helped.
